xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-tests/src/test/it/org/xwiki/flamingo/test/ui/XARExportTest.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.flamingo.test.ui;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.xwiki.flamingo.sking.test.po.ExportModal;
+import org.xwiki.flamingo.sking.test.po.OtherFormatPane;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.ui.AbstractTest;
+import org.xwiki.test.ui.po.ViewPage;
+import org.xwiki.tree.test.po.TreeElement;
+import org.xwiki.tree.test.po.TreeNodeElement;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Verify that the XAR export features works fine.
+ *
+ * @version $Id$
+ * @since 10.9RC1
+ */
+public class XARExportTest extends AbstractTest
+{
+    @Test
+    public void scenarioExportXAR() throws Exception
+    {
+        setupPages();
+        exportXARAll();
+        exportXARLotOfSelectedFiles();
+        exportXARWithUnselect();
+    }
+
+    private void setupPages() throws Exception
+    {
+        getUtil().loginAsSuperAdmin();
+
+        // Create a space Foo
+        getUtil().createPage("Foo", "WebHome", "Foo", "Foo");
+
+        // Create 100 pages under that space (will be used for the test with lot of selected pages)
```

Comment:
Do you need that many pages? Each page being saved takes some time AFAIK, unless you can do it in batch or in a database session.

Reply:
I'm using the REST API here so it does not take so much time (the test were taking around 20 seconds, once launched).
I need lot of pages to check the behaviour with a long list of pages selected, and I also want to check the "more pages" node, but indeed 100 is quite a lot. I'll create 50.

Reply:
It seems that you haven't updated the comment (and it's great that the pages creation is fast enough!).